### PR TITLE
Add new development tools

### PR DESCRIPTION
**Pull request by JoosuaKoskinen**
#### What are the main changes in this PR?

- Add all required tools as composer dependencies (+ Rector as new)
- Run Rector on whole project

**NOTE:** Rector recommends having PHP extensions `intl` and `sqlite`.

#### Where should a reviewer start?

Testing this must be thought a bit, there's a lot of changes.

#### Manual testing steps?
- `composer install`
- `composer test`
- `composer fix`?

### CODE REVIEW

**Review by Moppa5 — APPROVED** [review]
Looks good! I didn't find any major inconveniences or bugs. I couldn't test domain page very extensively as I don't have any domains and can't currently add any.

I got some PHP Warnings but these have been wondering around a while sometimes so I don't think it's a bug in this PR.
`[09-Jun-2021 11:15:13] WARNING: [pool www] child 63913, script '/data/wordpress/htdocs/wordpress/wp-admin/admin-ajax.php' (request: "POST /wordpress/wp-admin/admin-ajax.php") executing too slow (10.365692 sec), logging`

To sum up
- Documentation looks good and clear regarding the 
- Seravo Plugin features etc. work normally
- Statically analyzed the code & couldn't find anything odd
- Tested PHP 7.2 & current PHP 8.0 (I don't think there are significant stuff regarding PHP 7.3 or 7.4 if 7.2 works)
